services/w3c/w3c-validation.service.js

```diff
@@ -0,0 +1,269 @@
+        } else if (decodedPreset.match(/RDFa Lite 1\.1/i)) {
+          schema.push('http://s.validator.nu/html5-rdfalite.rnc')
+        } else {
+          schema.push('http://s.validator.nu/html5.rnc')
+        }
+      } else if (decodedPreset.match(new RegExp(xhtmlExpression, 'i'))) {
+        if (decodedPreset.match(/RDFa Lite 1\.1/i)) {
+          schema.push('http://s.validator.nu/xhtml1-ruby-rdf-svg-mathml.rnc')
+        } else if (decodedPreset.match(/1\.0 Strict, URL, Ruby, SVG 1\.1/i)) {
+          schema.push('http://s.validator.nu/xhtml1-ruby-rdf-svg-mathml.rnc')
+        } else {
+          schema.push('http://s.validator.nu/xhtml5.rnc')
+        }
+      } else if (decodedPreset.match(new RegExp(svgExpression, 'i'))) {
+        schema.push('http://s.validator.nu/svg-xhtml5-rdf-mathml.rnc')
+      }
+      schema.push('http://s.validator.nu/html5/assertions.sch')
+      schema.push('http://c.validator.nu/all/')
+    }
+    return schema.join(' ')
+  }
+
+  getMessage(messageTypes) {
+    const messageTypeKeys = Object.keys(messageTypes)
+    messageTypeKeys.sort() // Sort to make the order error, warning for display
+
+    if (Object.keys(messageTypes).length === 0) {
+      return 'validated'
+    }
+
+    const messages = messageTypeKeys.map(
+      key => `${messageTypes[key]} ${key}${messageTypes[key] > 1 ? 's' : ''}`
+    )
+    return messages.join(', ')
+  }
+
+  getColor(messageTypes) {
+    if ('error' in messageTypes) {
+      return 'red'
+    }
+
+    if ('warning' in messageTypes) {
+      return 'yellow'
+    }
+
+    return 'brightgreen'
+  }
+
+  getUrl(doc, preset, parser) {
+    const qs = { doc, out: 'json' }
+    if (preset) qs['schema'] = this.getSchema(preset)
+    if (parser) qs['parser'] = parser
+    return {
+      url: 'http://validator.nu',
+      schema,
+      options: {
+        qs,
+      },
+    }
+  }
+
+  async fetch(jsonUrl) {
+    return this._requestJson(jsonUrl)
+  }
+
+  async handle(_routeParams, { doc, preset, parser }) {
+    const jsonUrl = this.getUrl(doc, preset, parser)
+    const data = await this.fetch(jsonUrl)
+    const reducer = (accumulator, message) => {
+      let { type } = message
+      if (type === 'info') {
+        type = 'warning'
+      } else {
+        // All messages are suppose to have a type and there can only be info, error or non-document
+        // If a new type gets introduce this will flag them as errors
+        type = 'error'
+      }
+
+      if (!(type in accumulator)) {
+        accumulator[type] = 0
+      }
+      accumulator[type] += 1
+      return accumulator
+    }
+    const messageTypes = data.messages.reduce(reducer, {})
+    const badge = this.constructor.render({
+      message: this.getMessage(messageTypes),
+      color: this.getColor(messageTypes),
+    })
+    badge.link = [
+      url
+        .format({
+          host: jsonUrl.url,
+          query: jsonUrl.options.qs,
+        })
+        .toString(),
+    ]
```

Comment:
This is only for `social` type badges, so let's remove this
```suggestion
```

Reply:
Removed